### PR TITLE
implemented client encryption of refresh tokens

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/DSDEEPB-1863

stole the encryption module from thurloe
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,29 +1,9 @@
-# http://phusion.github.io/baseimage-docker/
-FROM phusion/baseimage
+FROM broadinstitute/scala-baseimage
```
Comment (dvoet) [inline Dockerfile:1]: switch to common base image, it has crypto extensions

File: Dockerfile [hunk 2]
```diff
@@ -1,29 +1,9 @@
...
-    apt-get update && \
-    apt-get install -y oracle-java8-installer && \
-    wget http://www.scala-lang.org/files/archive/scala-2.11.4.deb && \
-    wget http://dl.bintray.com/sbt/debian/sbt-0.13.7.deb && \
-    dpkg -i scala-2.11.4.deb && \
-    dpkg -i sbt-0.13.7.deb && \
-    apt-get update && \
-    apt-get install -y scala sbt && \
-    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /*.deb
-
```
Comment (dvoet) [inline Dockerfile:26]: these are in the common base image

File: build.sbt [hunk 3]
```diff
@@ -6,7 +6,7 @@ organization  := "org.broadinstitute"
 
 version       := "0.1"
 
-scalaVersion  := "2.11.2"
+scalaVersion  := "2.11.7"
```
Comment (dvoet) [inline build.sbt:9]: the common base image uses a later version of scala

File: docker/install.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@ set -e
 
 RAWLS_DIR=$1
 cd $RAWLS_DIR
-sbt assembly
+sbt -J-Xms4g -J-Xmx4g assembly
```
Comment (dvoet) [inline docker/install.sh:7]: needed more memory to build the docker image

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/PerRequest.scala [hunk 5]
```diff
@@ -18,6 +18,14 @@ import spray.http._
 import scala.language.postfixOps
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/PerRequest.scala:18]: changes in this class to make it work with scala 2.11.7

File: src/test/resources/application.conf [hunk 6]
```diff
@@ -22,3 +22,5 @@ submission-monitor-dispatcher {
   # Set to 1 for as fair as possible.
   throughput = 1
 }
+
+integration.runFullLoadTest = false
```
Comment (jmthibault79) [inline src/test/resources/application.conf:26]: Why did this change?

**Comment by jmthibault79** [comment]
:100:

File: src/test/resources/application.conf (hunk 6 above)
Comment (dvoet) [inline src/test/resources/application.conf:26]: it used to be there pre-vault, it makes the integration tests run

**Pull request merged**